### PR TITLE
write once and query bug fixes

### PR DESCRIPTION
**Pull request by gordonguthrie**
This PR also includes #1187, write once and query bug fixes.

Changes to querying to match the latest eleveldb API for querying.

### CODE REVIEW

**Comment by gordonguthrie** [comment]
It's passing test

File: src/riak_object.erl [hunk 1]
```diff
@@ -100,15 +99,15 @@
...
 
 %% @doc Constructor for new riak objects.
 -spec new(Bucket::bucket(), Key::key(), Value::value()) -> riak_object().
 new({T, B}, K, V) when is_binary(T), is_binary(B), is_binary(K) ->
     new_int({T, B}, K, V, no_initial_metadata);
 new(B, K, V) when is_binary(B), is_binary(K) ->
-    new_int(B, K, V, no_initial_metadata).
-
+    new_int(B, K, V, no_initial_metadata)
+.
```
Comment (gordonguthrie) [inline src/riak_object.erl]: full stop on wrong line

File: src/riak_object.erl [hunk 2]
```diff
@@ -1104,7 +1090,9 @@ decode_msgpack(ValBin) ->
 decode_maybe_binary(<<1, Bin/binary>>) ->
     Bin;
 decode_maybe_binary(<<0, Bin/binary>>) ->
-    binary_to_term(Bin).
+    binary_to_term(Bin);
+decode_maybe_binary(Bin) ->
+    decode_msgpack(Bin).
```
Comment (gordonguthrie) [inline src/riak_object.erl:1096]: Need to ensure that we review the stability of this change under this JIRA

https://bashoeng.atlassian.net/browse/RTS-333

Comment (andytill) [inline src/riak_object.erl:1096]: Fixed in e815bc5899c89a6f0811cb995bd3764020e1a41d

**Comment by gordonguthrie** [comment]
+1 e815bc5

**Comment by andytill** [comment]
@borshop merge

**Pull request merged**